solarkraft/test/e2e/verify_js_example.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * This is an example of how to write a Solarkraft runtime monitor in
+ * TypeScript.
+ *
+ * This is a monitor for the timelock contract from soroban-examples:
+ * https://github.com/stellar/soroban-examples/tree/v21.6.0/timelock
+ *
+ * We extend the SolarkraftJsMonitor class and provide methods corresponding to the
+ * contract functions. We use the `reverts_if` and `succeeds_with` methods to specify
+ * the conditions under which the contract should revert or succeed.
+ *
+ * The monitor is then verified by instantiating `TimelockMonitor` and calling the
+ * `verify` method with the transaction hashes of the transactions to be verified.
+ * We expect that the tranactions have been fetched from the ledger using the
+ * `solarkraft fetch` command.
+ *
+ * @author Thomas Pani, 2024
+ */
+import {
+    some,
+    every,
+    Env,
+    EnvRevert,
+    SolarkraftJsMonitor,
+    tokenTransferred,
+} from '../../src/verify_js.js'
+import _ from 'lodash'
+
+/* generate from contract spec */
+
+enum TimeBoundKind {
+    Before,
+    After,
+}
+
+interface TimeBound {
+    kind: TimeBoundKind
+    timestamp: number
+}
+
+interface ClaimableBalance {
+    token: string
+    amount: number
+    claimants: string[]
+    time_bound: TimeBound
+}
+
+/* the monitor */
+
+class TimelockMonitor extends SolarkraftJsMonitor {
+    deposit(
```

Comment:
I like this. One idea: When you write `deposit`, this may be confusing, as you do not describe the behavior of `deposit`. Would it make sense to have a name like `on_deposit`, `before_deposit`, or `deposit_ensures`?

Reply:
Good point! I think `on_deposit` would work well.

I've put it in #141 to follow up on.